### PR TITLE
report sys errors when failed to produce successfully

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

For python functions, when there are failures producing messages, the errors are not reported or log which makes it hard to determine if there is a problem

### CODE REVIEW

**Comment by merlimat** [comment]
run java8 tests

**Comment by jerrypeng** [comment]
run java8 tests

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java [hunk 1]
```diff
@@ -204,15 +225,17 @@ public void sendOutputMessage(TypedMessageBuilder<T> msg, Record<T> record)
             msg.sequenceId(record.getRecordSequence().get());
             msg.sendAsync()
                     .thenAccept(messageId -> record.ack())
+                    .exceptionally(getPublishErrorHandler(record))
```
Comment (merlimat) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java]:
```
CompletableFuture<MessageId> future = msg.sendAsync();

future.thenAccept(messageId -> record.ack())
      .exceptionally(getPublishErrorHandler(record))
future.join();
```

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by merlimat** [comment]
run java8 tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
ca070a4

**Pull request merged**